### PR TITLE
Integrate ido to mainnet runtime 

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR integrates ido to mainnet runtime

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,8 @@ panic = 'unwind'
 members = [
     'node',
     'runtime',
-    "pallets/pdex-migration"
+    'pallets/pdex-migration',
+    'pallets/Polkadex_Ido'
```
Comment (jangorecki) [inline Cargo.toml]: Is there any reason why not to align to naming convention that is already there in pallets when naming new one?

Comment (felixfaisal) [reply Cargo.toml]: Thanks for pointing it out @jangorecki

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -346,6 +347,8 @@ pub fn testnet_genesis(
 		"Total Supply Not equal to 20 million"
 	);
 	let vesting = get_vesting_terms();
+	let ether_address: H160 = "0xF59ae934f6fe444afC309586cC60a84a0F89Aaee".parse().unwrap();
+	let ether_asset_id: u128 = create_asset_id(ether_address);
```
Comment (Gauthamastro) [inline node/src/chain_spec.rs]: Remove these old constants.

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -394,9 +397,23 @@ pub fn testnet_genesis(
 		treasury: Default::default(),
 		orml_vesting: OrmlVestingConfig { vesting },
 		pdex_migration: PDEXMigrationConfig { max_tokens: ERC20_PDEX_SUPPLY, operational: false },
+		assets: AssetsConfig {
+			assets: vec![(ether_asset_id, root_key.clone(), false, 1)], /* FIXME What should be
+			                                                             * the min balance and
```
Comment (Gauthamastro) [inline node/src/chain_spec.rs]: No default configs for assets pallet.

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -394,9 +397,23 @@ pub fn testnet_genesis(
...
+			                                                             * is_sufficient? */
+			metadata: vec![],
+			accounts: vec![],
+		},
 	}
 }
 
+pub fn create_asset_id(token: H160) -> u128 {
+	let mut temp = [0u8; 16];
+	temp.copy_from_slice(&token[0..16]);
```
Comment (Gauthamastro) [inline node/src/chain_spec.rs]: Remove this.

File: pallets/polkadex-ido/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+hex = { version = "2.1.0", package = "rustc-hex", default-features = false }
+serde = { version = "1.0.127", optional = true }
+scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
+frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", default-features = false }
+frame-system = {  git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", default-features = false }
+sp-io = {   git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19",default-features = false }
+sp-core = {   git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", default-features = false }
+sp-std = {   git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", default-features = false }
+frame-benchmarking = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", optional = true }
+orml-tokens = {  git = "https://github.com/Polkadex-Substrate/open-runtime-module-library.git", default-features = false,  branch="polkadot-v0.9.19"}
```
Comment (Gauthamastro) [inline pallets/polkadex-ido/Cargo.toml]: We are not using orml dependencies anymore, please remove them

Comment (Gauthamastro) [reply pallets/polkadex-ido/Cargo.toml]: Also, remove any unused dependencies.

File: pallets/polkadex-ido/src/mock.rs [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+use super::*;
+
+use crate as polkadex_ido;
+use frame_support::PalletId;
+use frame_support::{traits::{Contains, Everything}};
+use frame_support::{parameter_types, traits::SortedMembers};
+use frame_support_test::TestRandomness;
+use frame_system::EnsureSignedBy;
+use orml_currencies::BasicCurrencyAdapter;
+use orml_traits::arithmetic::Zero;
```
Comment (Gauthamastro) [inline pallets/polkadex-ido/src/mock.rs]: Remove ORML stuff

File: runtime/src/lib.rs [hunk 7]
```diff
@@ -1208,6 +1267,9 @@ construct_runtime!(
 		// Pallets
 		OrmlVesting: orml_vesting::{Pallet, Storage, Call, Event<T>, Config<T>} = 28,
 		PDEXMigration: pdex_migration::pallet::{Pallet, Storage, Call, Event<T>, Config<T>} = 29,
+		RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Storage} = 32,
+		Assets: pallet_assets::{Pallet, Call, Storage, Event<T>, Config<T>} = 39,
```
Comment (Gauthamastro) [inline runtime/src/lib.rs]: re-order them as per index, please ;)

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/polkadex-ido/Cargo.toml (hunk 5 above)
Comment (jangorecki) [reply pallets/polkadex-ido/Cargo.toml]: https://crates.io/crates/cargo-udeps may be useful

Comment (felixfaisal) [reply pallets/polkadex-ido/Cargo.toml]: Thanks for suggesting @jangorecki . All dependencies are being used now as checked by `udeps` 
```
info: Loading save analysis from "/home/felix/pdex-code/fx-ido-refactor/Polkadex/target/debug/deps/save-analysis/libpallet_polkadex_ido_primitives-811fdb0155bdd19c.json"
info: Loading save analysis from "/home/felix/pdex-code/fx-ido-refactor/Polkadex/target/debug/deps/save-analysis/polkadex_ido-0596a546f6429521.json"
info: Loading save analysis from "/home/felix/pdex-code/fx-ido-refactor/Polkadex/target/debug/deps/save-analysis/libpolkadex_ido-d85e11832f3d92e7.json"
All deps seem to have been used.
```

File: runtime/src/lib.rs (hunk 7 above)
Comment (felixfaisal) [reply runtime/src/lib.rs]: Reordered

**Review by Gauthamastro — APPROVED** [review]
LGTM

**Review by zktony — APPROVED** [review]

**Comment by pavankanteti** [comment]
Closes #420

**Pull request merged**